### PR TITLE
rewrite using consistent style to other libraries

### PR DESCRIPTION
**Pull request by matin**
`stpmex` was the first client library that we wrote. I made some unusual choices because I didn't have much experience with soap APIs. With the experience from the other client libraries, I cleaned up the entire model that the library follows and make validation a lot cleaner in the process.

### CODE REVIEW

**Comment by pachCode** [comment]
Listo para revisarlo?

**Comment by matin** [comment]
@pachCode sí

File: stpmex/orden.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    cuentaBeneficiario: digits(10, 19)
+    institucionContraparte: digits(5, 5)
+    tipoCuentaBeneficiario: int
+
+    nombreOrdenante: Optional[truncated_str(39)] = None
+    cuentaOrdenante: Optional[digits(10, 19)] = None
+    institucionOperante: digits(5, 5) = STP_BANK_CODE
+    tipoCuentaOrdenante: Optional[int] = None
+
+    claveRastreo: truncated_str(29) = field(
```
Comment (pachCode) [inline stpmex/orden.py]: Si lo validas con truncated_str  y mandas la clave de rastreo como None manda error, si no lo envías en los argumentos si le asigna el valor default. No sé si ese sea el comportamiento deseado?

File: stpmex/orden.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        default_factory=lambda: f'CR{int(time.time())}'
+    )
+    referenciaNumerica: digits(1, 7) = field(
```
Comment (pachCode) [inline stpmex/orden.py]: Mismo caso que el de arriba

File: stpmex/orden.py (hunk 1 above)
Comment (matin) [reply stpmex/orden.py]: `pydantic` won't allow you to pass in `None` if the type specifically doesn't explicitly have `Optional[]`. However, the defaults still apply when the field is absent: https://github.com/cuenca-mx/stpmex-python/pull/39/commits/16075086b1812dd03ed8796331c6f5aaf1030d6c

File: stpmex/orden.py (hunk 2 above)
Comment (matin) [reply stpmex/orden.py]: addressed in other comment

**Pull request merged**